wasm/jit/jit_arm64.go

```diff
@@ -667,7 +667,20 @@ func (c *arm64Compiler) initializeReservedStackBasePointerRegister() {
 	calcStackBasePointerAddress.As = arm64.AADD
 	calcStackBasePointerAddress.To.Type = obj.TYPE_REG
 	calcStackBasePointerAddress.To.Reg = reservedRegisterForStackBasePointerAddress
-	calcStackBasePointerAddress.From.Type = obj.TYPE_SHIFT
-	calcStackBasePointerAddress.From.Offset = (int64(tmpReg)&31)<<16 | 0<<22 | (3&63)<<10
+	setShiftedRegister(calcStackBasePointerAddress, true, tmpReg, 3)
 	c.addInstruction(calcStackBasePointerAddress)
 }
+
+// setShiftedRegister modifies the given *obj.Prog so that .From or .To
+// becomes the "shifted register". For example, this is used to emit instruction like
+// "add  x1, x2, x3, lsl #3" which means "x1 = x2 + (x3 << 3)".
+// See https://github.com/twitchyliquid64/golang-asm/blob/v0.15.1/obj/link.go#L120-L131.
+func setShiftedRegister(inst *obj.Prog, onFrom bool, register int16, shiftNum int64) {
+	if onFrom {
```

Comment:
maybe var addr Addr... if onFrom add = inst.From else addr = inst.To then addr.Type=... that way no copy/paste

Reply:
removed onFrom in the first place as I realized that setting this to .To is illegal and we never encounter the usecase 😄